compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -2152,6 +2152,15 @@ class TrackingTypeComparer(initctx: Context) extends TypeComparer(initctx) {
         case _ =>
           cas
       }
+      def widenAbstractTypes(tp: Type) = new TypeMap {
+        def apply(tp: Type) = tp match {
+          case tp: TypeRef if tp.symbol.isAbstractOrParamType | tp.symbol.isOpaqueAlias => WildcardType
+          case tp: TypeVar if !tp.isInstantiated => WildcardType
+          case _: SkolemType | _: TypeParamRef => WildcardType
```

Comment:
I don't think SkolemTypes need to be mapped to WildcardTypes. It seems good enough to just do a
```
   apply(tp.info)
```
for them.

Reply:
Also, I think you need to follow alias types. Something like this:
```scala
case tp: TypeRef =>
    tp.info match {
      case TypeAlias(alias) => 
        val alias1 = widenAbstractTypes(alias)
        if (alias1 ne alias) alias1 else tp
      case _: TypeBounds => 
        WildcardType
  }
```
One could also consider mapping to a WildcardType with the underlying bounds widened recursively, but that's more complicated, and might not be necessary.

Reply:
I added the logic to follow aliases. If mapping over `.info` is good enough we can get rid of the `SkolemType` case since that's the default behavior of type maps.